### PR TITLE
Cache busting for icons & language files

### PR DESCRIPTION
**Pull request by bwindels**
- [x] cache busting for icons
- [x] cache busting for language files.

React-SDK PR: https://github.com/matrix-org/matrix-react-sdk/pull/2658

For i18n, I changed the copy-res.js script to generate content-hashed filenames and include those in languages.json, which then gets a content-hashed filename with file-loader.

Fixes https://github.com/vector-im/riot-web/issues/4136

### CODE REVIEW

**Comment by bwindels** [comment]
unit tests on react sdk are currently broken...

File: scripts/copy-res.js [hunk 1]
```diff
@@ -195,21 +163,28 @@ function genLangFile(lang, dest) {
 
     translations = weblateToCounterpart(translations);
 
-    fs.writeFileSync(dest + lang + '.json', JSON.stringify(translations, null, 4));
+    const json = JSON.stringify(translations, null, 4);
+    const jsonBuffer = Buffer.from(json);
+    const digest = loaderUtils.getHashDigest(jsonBuffer, "sha512", "base64", 7);
```
Comment (jryans) [inline scripts/copy-res.js]: I think `base64` includes `/`, which could get weird for things in file paths. Also, I'm not sure how much it actually impacts build time, but `sha512` seems like overkill for this problem... We really only need the simplest thing that gives a different value when the file changes. (Also we're only keeping 7 bytes of it, so powerful hashes like `sha512` are mostly doing work that is thrown out.)

For images via Webpack we're currently using `[hash:7]`, which [seems to equate to](https://github.com/webpack/loader-utils/blob/b91a76c0c98bf71df5a3a1c808ea86268c3716c1/lib/interpolateName.js#L96) `getHashDigest(buffer, null, null, 7)`, which will then default to `md5` and `hex`, which seems fast, safe, and sufficient for the problem here.

**Comment by aaronraimist** [comment]
Can this/should this be applied to config.json as well? https://github.com/vector-im/riot-web/issues/7241

**Comment by jryans** [comment]
> Can this/should this be applied to config.json as well? #7241

Hmm, I don't think we can use the same approach for config.json, as it's not part of Riot's build process. I don't think we would want to entangle it in the build process either, as people might use other configuration management systems (Ansible, etc.) to produce and manage the file.

**Pull request merged**